### PR TITLE
feat: initial jsii-srcmak work for golang

### PR DESCRIPTION
**Pull request by campionfellin**
Starting off with some `golang` stuff here... some interesting issues with the `moduleKey` that I haven't seen before.. it seems to be required input (rather than optional previously).

JSII I think gets the output directory from `name` here:

https://github.com/aws/jsii-srcmak/blob/37bc954923b8b5dd076eec8412ffe646c1dca4be/src/compile.ts#L54-L57

And in JSII:

https://github.com/aws/jsii/blob/32c0add5edd0ed57d535241b483168e2b7e731ce/packages/jsii-pacmak/lib/targets/go.ts#L146-L151

Which somewhat makes sense, but when using the `jsii-srcmak` CLI, you usually don't put in the `moduleKey` explicitly, ending with it being a random hash:

```
https://github.com/aws/jsii-srcmak/blob/master/src/compile.ts#L29
```

So my `dist` folder looks like:

```
dist/go/4c6b576fbe7e27053874813d9754cb2e46811d806c1e0aa9aae8add4c3763060
```

I think JSII should probably base the `outdir` on `moduleName` input... will poke around there to see what I can do

Signed-off-by: campionfellin <campionfellin@gmail.com>

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -87,6 +90,21 @@ async function main() {
...
   }
 
+  function parseGoLangOptions() {
+    const outdir = argv['golang-outdir'];
+    const module = argv['golang-module'];
+    if (!outdir && !module) { return undefined; }
+    if (!outdir) { throw new Error('--golang-outdir is required'); }
+    if (!module) { throw new Error('--golang-module is required'); }
+    return {
+      // moduleKey: module,
```
Comment (campionfellin) [inline src/cli.ts:100]: This is where you could explicitly set `moduleKey` for `golang` but then it breaks other languages

Comment (ansgarm) [reply src/cli.ts:100]: Hey @campionfellin, could you elaborate on how it does break the other languages. I tried it locally and it seemed to still work. I'd think it shouldn't matter to JSII whether there's a random hash or some actual name if the result does not contain that hash/name anyway? Did I miss something?

Comment (campionfellin) [reply src/cli.ts:100]: I can't quite remember here... I think it had something to do with `moduleKey` needing to be a random hash for some languages, but needed to be named for `go`. Perhaps I just couldn't figure it out at the time, but now it sounds like it's working for you!

**Comment by eladb** [comment]
Superseded by #476

**Pull request closed without merging**